### PR TITLE
ci: Add static checks for anchor

### PR DESCRIPTION
**Pull request by wzshiming**
ci: Add static checks for anchor 

Only this uses EOT, the others are EOF, uniformly changed to EOF to
avoid confusion

Fixes: #4102

Signed-off-by: Shiming Zhang <wzshiming@foxmail.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Thanks @wzshiming. The "fixes" comment must be in the commit though and specifying a number, not a URL. Please see:

- https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#closing-issues

**Comment by wzshiming** [comment]
> Thanks @wzshiming. The "fixes" comment must be in the commit though and specifying a number, not a URL. Please see:
> 
> * https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#closing-issues

Updated. I sent an issue with a link to the original discussion.

File: .ci/static-checks.sh [hunk 1]
```diff
@@ -823,11 +823,28 @@ static_check_docs()
...
 
 	[ $docs_failed -eq 0 ] || die "spell check failed, See https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#spelling for more information."
 }
 
+static_check_eof()
+{
+	local file="$1"
+	local anchor="EOF"
+
+	local invalid=$(bash "${doc_to_script_cmd}" -s "$file" - |\
```
Comment (wzshiming) [inline .ci/static-checks.sh:838]:
```suggestion
	local invalid=$(cat "$file" |\
```

Comment (wzshiming) [reply .ci/static-checks.sh]: @jodh-intel 
I debugged it seems to be this error.

File: .ci/static-checks.sh [hunk 2]
```diff
@@ -1060,6 +1077,8 @@ static_check_shell()
 		{ $chronic bash -n "$script"; ret=$?; } || true
 
 		[ "$ret" -eq 0 ] || die "check for script '$script' failed"
+
+		static_check_shell "$script"
```
[the remark this answers is not in the document]
Comment (jodh-intel) [reply .ci/static-checks.sh:1081]: Any thoughts on this @wzshiming?

File: .ci/static-checks.sh [hunk 3]
```diff
@@ -823,11 +823,28 @@ static_check_docs()
...
 	done
 
 	[ $docs_failed -eq 0 ] || die "spell check failed, See https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#spelling for more information."
 }
 
+static_check_eof()
+{
+	local file="$1"
+	local anchor="EOF"
+
```
Comment (wzshiming) [inline .ci/static-checks.sh:837]:
```suggestion
	[ -z "$file" ] && return 0
```
CI's error message This file may be empty

File: .ci/static-checks.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (wzshiming) [reply .ci/static-checks.sh]: It looks like `static_check_eof` ,Updated , Thank you

**Comment by jodh-intel** [comment]
Hi @wzshiming - do you still have time to work on this PR, or shall we close it?

**Comment by wzshiming** [comment]
> Hi @wzshiming - do you still have time to work on this PR, or shall we close it?

Yes, I have changed it last time, need test.

**Review by jodh-intel — APPROVED** [review]
Thanks @wzshiming !

lgtm

**Comment by wzshiming** [comment]
It's not clear why it failed at this PR, and there are no error messages.
I run ci in my forked repository, and it seems to be fine. [wzshiming/kata-tests#1](https://github.com/wzshiming/kata-tests/pull/1)
Could you have time to take a look?

**Comment by wzshiming** [comment]
I sent a PR to migrate the current repo anchor to EOF
https://github.com/kata-containers/tests/pull/4238

**Comment by wzshiming** [comment]
I changed CI in my forkd repository and finally passed
[wzshiming/kata-tests#1](https://github.com/wzshiming/kata-tests/pull/1)

**Comment by wzshiming** [comment]
https://github.com/wzshiming/kata-tests/runs/4406867186?check_suite_focus=true

The Workflow in my repo looks like it passed the check

**Comment by wzshiming** [comment]
@jodh-intel 
Need to remove do-not-merge and test

**Comment by jodh-intel** [comment]
One more ack needed to land this PR folks.

/test

**Review by fidencio — APPROVED** [review]
lgtm, thanks @wzshiming!

**Pull request merged**